core/src/utilities/helpers/global-search-helpers.js

```diff
@@ -1,13 +1,195 @@
 /* istanbul ignore file */
-class GlobalSearchHelperClass {
-  constructor() {}
+import { GenericHelpers } from './';
+import { KEYCODE_ARROW_UP, KEYCODE_ARROW_DOWN, KEYCODE_ENTER, KEYCODE_ESC } from './../keycode.js';
+import { Routing } from './../../services/routing';
+import { LuigiI18N } from './../../core-api';
+export class GlobalSearchHelperClass {
+  dispatch;
+  search;
+  isCustomSearchRenderer;
+  isCustomSearchResultItemRenderer;
+  luigiCustomSearchItemRenderer__slotContainer;
+
+  constructor(search, dispatcher) {
+    this.search = search;
+    this.dispatch = dispatcher;
+  }
+
+  getCustomRenderer() {
+    if (!this.search.searchProvider) return;
+    this.isCustomSearchRenderer = GenericHelpers.isFunction(this.search.searchProvider.customSearchResultRenderer);
+    this.isCustomSearchResultItemRenderer = GenericHelpers.isFunction(
+      this.search.searchProvider.customSearchResultItemRenderer
+    );
+  }
+
+  updateLuigiCustomSearchItemRenderer__slotContainer(updatedSlotContainer) {
+    this.luigiCustomSearchItemRenderer__slotContainer = updatedSlotContainer;
+  }
 
   handleVisibilityGlobalSearch() {
     if (!document.querySelector('.lui-global-search')) return;
     const globalSearchCtn = document.querySelector('.lui-global-search');
     const condition = globalSearchCtn.offsetWidth <= 384;
     globalSearchCtn.classList.toggle('lui-global-search-toggle', condition);
   }
-}
 
-export const GlobalSearchHelper = new GlobalSearchHelperClass();
+  setSearchPlaceholder(inputElement) {
+    const placeHolder = this.getSearchPlaceholder();
+    if (placeHolder) {
+      inputElement.placeholder = placeHolder;
+    }
+  }
+
+  getSearchPlaceholder() {
+    const searchProvider = this.search.searchProvider;
+    if (!searchProvider || !searchProvider.inputPlaceholder) {
+      return undefined;
+    }
+    const currentLocale = LuigiI18N.getCurrentLocale();
+    if (GenericHelpers.isFunction(searchProvider.inputPlaceholder)) {
```

Comment:
For multiple conditions switch statement seems to be a better choice

Reply:
I can change that to switch, but I don't see much benefit from that.
@JohannesDoberer Please advise which approach do you prefer :)

Reply:
OK - let's leave it as is